component/metrics/mutate/mutate.go

```diff
@@ -101,19 +102,21 @@ func (c *Component) Update(args component.Arguments) error {
 // series. This is a blocker for releasing a production-grade  of the metrics.mutate
 // component.
 func (c *Component) Receive(ts int64, metricArr []*metrics.FlowMetric) {
-	app := c.appendable.Appender(context.Background())
+	c.mut.Lock()
+	defer c.mut.Unlock()
+
+	relabelledMetrics := make([]*metrics.FlowMetric, 0)
 	for _, m := range metricArr {
-		m.Labels = relabel.Process(m.Labels, c.mrc...)
-		if m.Labels == nil {
+		newLabels := relabel.Process(m.LabelsCopy(), c.mrc...)
+		if newLabels == nil {
 			continue
 		}
-		_, err := app.Append(storage.SeriesRef(m.GlobalRefID), m.Labels, ts, m.Value)
-		if err != nil {
-			level.Error(c.opts.Logger).Log("msg", "failed to forward sample from metrics.mutate component", "err", err, "componentID", c.opts.ID)
-		}
+		relabelledMetrics = append(relabelledMetrics, metrics.NewFlowMetric(metrics.GlobalRefMapping.GetOrAddGlobalRefID(newLabels), newLabels, m.Value()))
```

Comment:
nit: kind of a long line, can we assign the new metric to a variable rather than constructing it in the append?